metadata-io/src/main/java/com/linkedin/metadata/search/transformer/SearchDocumentTransformer.java

```diff
@@ -210,8 +210,13 @@ public void setSearchableValue(
                     fieldName,
                     JsonNodeFactory.instance.booleanNode((Boolean) firstValue.orElse(false)));
               } else {
-                searchDocument.set(
-                    fieldName, JsonNodeFactory.instance.booleanNode(!fieldValues.isEmpty()));
+                final boolean hasValue;
+                if (DataSchema.Type.STRING.equals(valueType)) {
```

Comment:
If the string field does not have `hasValuesFieldName` defined in searchable annotation pdl, then for search, we are dependent on the value present in the field in ES.

My field looks like:
```
@Searchable = {
    "fieldType": "TEXT"
  }
  description: optional string
```

But the value in ES, is not updated to empty string or null. If the entity existed before with a valid description. And then updated to empty description, through UI or ingestion.

I think the expectation after updating the entity description from some existing value to empty in ES will be empty or null. (maybe will contain the same result as in DB) in such cases.